doc/examples_sphinx-gallery/stochastic_variability.py

```diff
@@ -0,0 +1,93 @@
+"""
+.. _tutorials-stochastic-variability:
+
+=========================================================
+Stochastic Variability in Community Detection Algorithms
+=========================================================
+
+This example demonstrates the variability of stochastic community detection methods by analyzing the consistency of multiple partitions using similarity measures (NMI, VI, RI) on both random and structured graphs.
+
+"""
+# %%
+# Import Libraries
+import igraph as ig
+import numpy as np
+import matplotlib.pyplot as plt
+import itertools
+
+# %%
+# First, we generate a graph.
+# Generates a random Erdos-Renyi graph (no clear community structure)
+def generate_random_graph(n, p):
+    return ig.Graph.Erdos_Renyi(n=n, p=p)
+  
+# %%
+# Generates a clustered graph with clear communities using the Stochastic Block Model (SBM)
+def generate_clustered_graph(n, clusters, intra_p, inter_p):
+    block_sizes = [n // clusters] * clusters
+    prob_matrix = [[intra_p if i == j else inter_p for j in range(clusters)] for i in range(clusters)]
+    return ig.Graph.SBM(sum(block_sizes), prob_matrix, block_sizes)
+
+# %%
+# Computes pairwise similarity (NMI, VI, RI) between partitions
+def compute_pairwise_similarity(partitions, method):
+    """Computes pairwise similarity measure between partitions."""
+    scores = []
+    for p1, p2 in itertools.combinations(partitions, 2):
+        scores.append(ig.compare_communities(p1, p2, method=method))
+    return scores
+
+# %%
+# Stochastic Community Detection
+# Runs Louvain's method iteratively to generate partitions
```

Comment:
This is called the [Louvain](https://en.wikipedia.org/wiki/UCLouvain) method, not Louvain's method.

Can you include a short explanation of why the result is different on each run? This is often a point of confusion for empirical researchers who are inexperienced in data analysis.

This is a modularity maximization method. Since the exact maximization of modularity is NP-hard, the Louvain method uses a greedy heuristic, processing vertices in a random order.